core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2690,6 +2690,46 @@ public void updateMount(AlluxioURI alluxioPath, MountContext context)
     }
   }
 
+  /**
+   * Mounts an under storage path to an Alluxio path.
+   * <p>
+   * Links an under storage path to an Alluxio path,
+   * where files and folders created in Alluxio space
+   * under the path will be backed by a corresponding
+   * file or folder in the under storage path.
+   * <p>
+   * Increments {@link DefaultFileSystemMaster.Metrics#MOUNT_OPS},
+   * which keeps track of the number of mount operations made. Uses
+   * {@link LockingScheme} to determine that the path should be locked
+   * with the lock pattern {@link LockPattern#WRITE_EDGE}.
+   * <p>
+   * Attempts to instantiate {@link RpcContext} through {@link DefaultFileSystemMaster#createRpcContext()}
+   * and create an object of type {@link LockedInodePath} by locking all inodes inside of
+   * {@link DefaultFileSystemMaster#mInodeTree} with {@link InodeTree#lockInodePath}
+   * in order to avoid deadlocks.
+   * <p>
+   * Attempts to check whether the permission to {@link Mode.Bits#WRITE} is enabled.
+   * If it is not enabled, an exception in thrown and the UFS path is not mounted to
+   * Alluxio. Otherwise, checks whether the UFS should be synchronized, in which case
+   * {@link DefaultFileSystemMaster#syncMetadata} is invoked.
+   * <p>
+   * Mounts the UFS path internally using {@link DefaultFileSystemMaster#mountInternal}, providing:
+   *                    1) rpcContext, the RpcContext;
+   *                    2) inodePath, the LockedInodePath;
+   *                    3) ufsPath, the provided UFS path to mount to Alluxio;
+   *                    4) context, the provided object of type {@link MountContext},
+   *                    used to merge and wrap {@link MountPOptions}.
+   *
+   * @param alluxioPath the Alluxio path to mount to
+   * @param ufsPath     the UFS path to mount
+   * @param context     the mount context
+   * @throws FileAlreadyExistsException If an Alluxio path already exists.
```

Comment:
Typically, the convention in Alluxio is to not javadoc these exceptions, since the exceptions are usually self-explanatory.